docs/erc1155-skills.md

```diff
@@ -0,0 +1,214 @@
+with a one-step atomic swap functionality. That means the trade between the two AEAs can be trustless.
+
+### Launch an OEF node
+In a separate terminal, launch a local OEF node (for search and discovery).
+``` bash
+python scripts/oef/launch.py -c ./scripts/oef/launch_config.json
+```
+
+Keep it running for all the following demos.
+
+## Demo: Ledger payment
+
+A demo to run a scenario with a true ledger transaction on Ethereum `ropsten` network. This demo assumes the client
+does not trust the deployer AEA to make the trade.
+
+### Create the deployer AEA (ledger version)
+
+Create the AEA that will deploy the contract.
+
+``` bash
+aea create my_erc1155_deploy
+cd my_erc1155_deploy
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/erc1155_deploy:0.1.0
+aea add contract fetchai/erc1155:0.1.0
+aea install
+```
+Additionally, create the private key for the deployer AEA.
+
+Generate and add a key for Ethereum use:
+``` bash
+aea generate-key ethereum
+aea add-key ethereum eth_private_key.txt
+```
+
+
+### Create the client AEA (ledger version)
+
+In another terminal, create the AEA that will sign the transaction.
+
+``` bash
+aea create my_erc1155_client
+cd my_erc1155_client
+aea add connection fetchai/oef:0.1.0
+aea add skill fetchai/erc1155_client:0.1.0
+aea add contract fetchai/erc1155:0.1.0
+aea install
+```
+
+Additionally, create the private key for the client AEA.
+
+Generate and add a key for Ethereum use:
+```bash
+aea generate-key ethereum
+aea add-key ethereum eth_private_key.txt
+```
+
+### Update the AEA configs
+
+Both in `my_erc1155_deploy/aea-config.yaml` and
+`my_erc1155_client/aea-config.yaml`, replace `ledger_apis: {}` with the following based on the network you want to connect
+
+Connect to Ethereum:
+``` yaml
+ledger_apis:
+  ethereum:
+    address: https://ropsten.infura.io/v3/f00f7b3ba0e848ddbdc8941c527447fe
+    chain_id: 3
+    gas_price: 50
+```
+
+### Update the deployer AEA skill configs
+
+In `my_erc1155_deploy/vendor/fetchai/skills/erc1155_deploy/skill.yaml`, update the details based on the following:
+``` yaml
+name: erc1155_deploy
+author: fetchai
+version: 0.1.0
+license: Apache-2.0
+fingerprint: {}
+description: "The erc1155 deploy skill implements the functionality to depoly and interact with a smart contract."
+contracts: ['fetchai/erc1155:0.1.0']
+behaviours:
+  service_registration:
+    class_name: ServiceRegistrationBehaviour
+    args:
+      services_interval: 60
+handlers:
+  default:
+    class_name: FIPAHandler
+    args: {}
+  transaction:
+    class_name: TransactionHandler
+    args: {}
+models:
+  strategy:
+    class_name: Strategy
+    args:
+      ledger_id: 'ethereum'
+      is_ledger_tx: True
+      nft: 1
+      ft: 2
+      item_ids: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+      mint_stock: [100, 100, 100, 100, 100, 100, 100, 100, 100, 100]
+      from_supply: 10
+      to_supply: 0
+      value: 0
+      search_schema:
```

Comment:
This search schema seems incorrect?

Reply:
This is the default schema and we are using only to find each other.

Reply:
There is no such thing as a default schema. You can define a new one here. just provide different values?